### DISCUSSION TITLE
"Duplicate definition of column" MappingExceptions in class inheritance

### DISCUSSION THREAD

**Discussion by 99hops** [doctrine/doctrine2#11040] [Support Questions]
Hello, setting `$reportFieldsWhereDeclared=true` as per https://github.com/doctrine/orm/pull/10454 https://github.com/doctrine/orm/pull/10455 results in ```Duplicate definition of column 'id' on entity 'App\Entity\Backup' in a field or discriminator column mapping.``` for no obvious reason.

```
#[ORM\Entity]
class Backup extends Snapshot
{
}
```
```
#[ORM\Entity]
#[ORM\InheritanceType('SINGLE_TABLE')]
#[ORM\DiscriminatorColumn(name: 'type', type: 'integer', columnDefinition: 'integer DEFAULT 0')]
#[ORM\DiscriminatorMap([
   0 => Snapshot::class,
   1 => Backup::class
])]
class Snapshot extends SnapshotMappedSuperclass
{
}
```
```
#[ORM\MappedSuperclass]
#[ORM\Table('snapshot')]
class SnapshotMappedSuperclass
{
    #[ORM\Id]
    #[ORM\Column(type: 'uuid')]
    protected Uuid $id;
}
```

What is the point of this exception in the given context? It seems unreasonable to have it? Now problem is Snapshot entity contains a lot of relations, metadata, etc that needs to be duplicated in every instance extending SnapshotEntity to avoid this exception which makes no sense to me..

Please advice (:
Thanks! @mpdude

**Comment by mpdude** [comment]
`Snapshot` and `Backup` are tagged as entities as well, right? Your code example does not show that.

**Reply by 99hops** [reply]
Hi, thanks for jumping in, yes they are `#[ORM\Entity]`

**Reply by mpdude** [reply]
Ok, next question: Does `Table` make sense on the MappedSuperclass?

**Reply by mpdude** [reply]
I am unable to reproduce this. Does it also happen with annotations instead of attributes?

Can you simplify this as much as possible (remove UUID, do not specify column/table names etc.) and see if and what makes the error go away?

**Reply by 99hops** [reply]
@mpdude I was able to reproduce this with minimal setup [https://github.com/nforced/orm-mapping](https://github.com/nforced/orm-mapping). It turns out this is somehow related to the fact I use bundle for the MappedSuperclass.. Please tell me what do you think about it? Thanks!

**Reply by 99hops** [reply]
I'll try to debug it a bit in the weekend

**Comment by Cartman34** [comment]
I am having some trouble like this with Doctrine 3.
